StRoot/StETofUtil/StETofGeometry.cxx

```diff
@@ -81,25 +75,54 @@ StETofNode::StETofNode( const TGeoPhysicalNode& gpNode, const float& dx, const f
   mDebug( false )
 {
     mGeoMatrix = static_cast< TGeoHMatrix* > ( gpNode.GetMatrix() );
-    /*
-    double* trans  = mGeoMatrix->GetTranslation();
-    double* rot    = mGeoMatrix->GetRotationMatrix();
-
-    LOG_INFO << trans[0] << "  " << trans[1] << "  " << trans[2] << endm;
-
-    LOG_INFO << rot[0] << "  " << rot[1] << "  " << rot[2] << endm;
-    LOG_INFO << rot[3] << "  " << rot[4] << "  " << rot[5] << endm;
-    LOG_INFO << rot[6] << "  " << rot[7] << "  " << rot[8] << endm;
-    */
     mBox = static_cast< TGeoBBox* > ( gpNode.GetShape() );
 
     // resize mBox with dx and dy
     float dz = mBox->GetDZ();
-    mBox->SetBoxDimensions( dx, dy, dz );
+    mBox->SetBoxDimensions( dx, dy, dz ); 
 
     buildMembers();
 }
 
+StETofNode::StETofNode( const TGeoPhysicalNode& gpNode, const float& dx, const float& dy, const StThreeVectorD alignment )
+: mSafetyMarginX( 0. ),
+  mSafetyMarginY( 0. ), 
+  mDebug( false )
+{
+
+    mGeoMatrix = static_cast< TGeoHMatrix* > ( gpNode.GetMatrix() );
+
+    mBox = static_cast< TGeoBBox* > ( gpNode.GetShape() );
+
+    // resize mBox with dx and dy
+    float dz = mBox->GetDZ();
+    mBox->SetBoxDimensions( dx, dy, dz ); 
+
+    //  modified  buildMembers() method. PW;
+    // build member variables: mMinEta, mMaxEta, mMinPhi, mMaxPhi, mCenter, mNormal 
+    double xl[3] = { -alignment.x(), -alignment.y(), -alignment.z() };
+//z alignment is positive to reflect the correct direction in which z has to be moved in global coordinates. x and y are negative to match the numbers from hit-intersection 
```

Comment:
I removed this comment since it is not up to date anymore. All alignment numbers are passed as negatives, so the comment about z being positive doesn't make sense any more.